FromZeroToHero.sln

```diff
@@ -1,15 +1,177 @@
+        {
+            int[] newNumbers = new int[numbers.Length + 1];
+            int index = numbers.Length;
+            for (int i = 0; i < numbers.Length + 1; i++)
+            {
+                if (i == index)
+                {
+                    newNumbers[i] = newNumber;
+                }
+                else
+                {
+                    newNumbers[i] = numbers[i];
+                }
+            }
+            Console.WriteLine();
+            Console.WriteLine("Add element at the end of array");
+            for (int i = 0; i < numbers.Length + 1; i++)
+            {
+                Console.Write(newNumbers[i] + " ");
+            }
+        }
+
+        private static void AddElementAtStart(int[] numbers, int newNumber)
+        {
+            int[] newNumbers = new int[numbers.Length + 1];
+            int index = 0;
+            for (int i = 0; i < numbers.Length + 1; i++)
+            {
+                if (i == index)
+                {
+                    newNumbers[i] = newNumber;
+                }
+                else
+                {
+                    newNumbers[i] = numbers[i - 1];
+                }
+            }
+            Console.WriteLine();
+            Console.WriteLine("Add element at the beginning of array");
+            for (int i = 0; i < numbers.Length + 1; i++)
+            {
+                Console.Write(newNumbers[i] + " ");
+            }
+        }
+
+        private static void AddElementAtAnyPosition(int[] numbers, int newNumber, int position)
+        {
+            int[] newNumbers = new int[numbers.Length + 1];
+            for (int i = 0; i < numbers.Length + 1; i++)
+            {
+                if (i == position)
+                {
+                    newNumbers[i] = newNumber;
+                }
+                else if (i < position)
+                {
+                    newNumbers[i] = numbers[i];
+                }
+                else
+                {
+                    newNumbers[i] = numbers[i - 1];
+                }
+            }
+            Console.WriteLine();
+            Console.WriteLine("Add element at the the position you want in the array");
+            for (int i = 0; i < numbers.Length + 1; i++)
+            {
+                Console.Write(newNumbers[i] + " ");
+            }
+        }
+        private static void RemoveElementAtStart(int[] numbers)
+        {
+            int[] newArray = new int[numbers.Length - 1];
+            for (int i = 0; i < numbers.Length - 1; i++)
+            {
+                newArray[i] = numbers[i + 1];
+            }
+            Console.WriteLine();
+            Console.WriteLine("Remove first element in array");
+            for (int i = 0; i < numbers.Length - 1; i++)
+            {
+                Console.Write(newArray[i] + " ");
+            }
+        }
+
+        private static void RemoveElementAtEnd(int[] numbers)
+        {
+            int[] newArray = new int[numbers.Length - 1];
```

Comment:
Very consistent and precise array methods here.  One thing you might consider is reusing your general purpose methods (e.g. `RemoveElementAtPosition`) to implement your special purpose methods (e.g. `RemoveElementAtEnd`).